pkg/querier/worker/scheduler_processor.go

```diff
@@ -138,8 +128,16 @@ func (sp *schedulerProcessor) querierLoop(c schedulerpb.SchedulerForQuerier_Quer
 				logger = util_log.WithContext(ctx, sp.log)
 			)
 
+			start := time.Now()
+			tenant, _ := tenant.TenantID(ctx)
+			sp.metrics.inflightRequests.Inc()
+			level.Debug(logger).Log("msg", "tracking inflight request", "tenant", tenant, "op", "enqueue")
+
 			sp.runRequest(ctx, logger, request.QueryID, request.FrontendAddress, request.StatsEnabled, request.HttpRequest)
 
+			sp.metrics.inflightRequests.Dec()
```

Comment:
do we need to be worried about any conditions where the code would return without this being caused? I couldn't find anything obvious, so probably just a panic which would bring down the pod anyway, but figured I'd ask.

Reply:
Yeah, I believe only a panic can stop this code path from executing here.